numpy/ma/core.pyi

```diff
@@ -371,7 +374,13 @@ class _extrema_operation(_MaskedUFunc):
     def reduce(self, target, axis=...): ...
     def outer(self, a, b): ...
 
-def min(obj, axis=..., out=..., fill_value=..., keepdims=...): ...
+def min(
+    obj: MaskedArray[_ShapeLike, _SCT] | _ArrayLike[_SCT],
+    axis: None | _ShapeLike = ...,
+    out: None = ...,
+    fill_value: None | _ScalarLike_co = ...,
+    keepdims: None | bool = ...,
+) -> _SCT: ...
```

Comment:
if `axis` is `_ShapeLike` and not `None`, the return type will be the same type as `obj`, not `_SCT`.